eclair-core/src/main/scala/fr/acinq/eclair/Eclair.scala

```diff
@@ -64,6 +68,10 @@ object TimestampQueryFilters {
   }
 }
 
+object SignedMessage {
+  def signedBytes(message: String): ByteVector32 = Crypto.hash256(ByteVector(("Lightning Signed Message:" ++ message).getBytes))
```

Comment:
I think it's misleading to have `message` be a `String` here, what encoding should it be using?
If it's base64 we can't simply do `message.getBytes`.
This is why my suggestion was to use `def signedBytes(message: ByteVector)` and deal with the encoding (base64) outside of this method.

Reply:
I don't think the format of the message is that much important. If we receive base64 or hexadecimal data we just encode the ASCII representation of that data.

I don't see why this command:
```
eclair-cli signmessage --msg=$(echo -n "I was here" | base64)
```

should be preferable to this:
```
eclair-cli signmessage --msg="I was here"
```

If someone wants to encode binaries data, they're free to use base64, hexadecimal or some exotic encoding like zbase32 for the ASCII representation.

Reply:
It is very important, otherwise we won't sign the same thing as lnd or c-lightning and won't be compatible.
*What you sign are bytes, not strings,* so you absolutely need to specify the encoding that converts between the two, otherwise the user doesn't know what is really signed.
Note that ASCII (or UTF8) _is_ an encoding, just like base64.

With your current code, the two commands you mention will **not** result in the same signature.
So how can the user know which one will match lnd and c-lightning?
We want the user to provide us the message as *bytes*, but since these are messy to handle in CLI/API we simply ask that these bytes be provided as a base64 string.

Reply:
Right now we sign the same thing as lnd and c-lightning.

This command:
```
eclair-cli signmessage --msg="I was here"
```
will produce a signature that lnd and c-lightning can verify like this:
```
lightning-cli checkmessage 'I was here' ${signature}
lncli verifymessage --msg 'I was here' --sig ${signature}
```

If I do what you asks, the user will need to use this command instead:
```
eclair-cli signmessage --msg=$(echo -n 'I was here' | base64)
```
It puts the burden on the users to remember to convert in base64 the message they want to sign.